tools/osbuild-depsolve-dnf5

```diff
@@ -131,14 +151,33 @@ class Solver():
         # Load the file lists, so dependency on paths will work
         conf.optional_metadata_types = ['comps', 'filelists']
 
-        # Cannot modify .conf() values after this
-        self.base.setup()
-
         try:
             # NOTE: With libdnf5 packages are excluded in the repo setup
             for repo in repos:
                 self._dnfrepo(repo, exclude_pkgs)
 
+            if root_dir:
+                # This sets the varsdir to ("{root_dir}/usr/share/dnf5/vars.d/", "{root_dir}/etc/dnf/vars/") for custom
+                # variable substitution (e.g. CentOS Stream 9's $stream variable)
+                conf.installroot = root_dir
+                conf.varsdir = (os.path.join(root_dir, "etc/dnf/vars"), os.path.join(root_dir, "usr/share/dnf5/vars.d"))
+
+                # NOTE: No combination of varsdir and installroot seems to be doing what we expect (see
+                # osbuild-depsolve-dnf and tests). Read them manually for now until we fix this.
```

Comment:
In dnf4, we have:
```python
base.conf.substitutions.update_from_etc(root_dir)
```

which will load any vars from `<root_dir>/etc/dnf/vars` and `<root_dir>/etc/yum/vars`.  This is verified in our test.
From the dnf5 docs, I got the impression that setting the installroot would make it load vars from `<installroot>/etc/dnf/vars` and `<installroot>/usr/share/dnf5/vars.d`, but this isn't working.

Reply:
I've looked at the dnf5 code, and I don't see any obvious bugs there. As far as I can tell, as long as `conf.use_host_config` is false (which is the default) it will prefix `reposdir` and `varsdir` with `installroot`.

Reply:
I think we can leave it as is for now.  I opened https://github.com/rpm-software-management/dnf5/issues/1374 and we can return to this when we figure out what's wrong.

Reply:
This has been fixed.